src/components/TodoList/List/TodoList.tsx

```diff
@@ -2,24 +2,31 @@ import { FC } from 'react';
 
 import { useAppSelector } from '@redux';
 
+import { useTodoTask } from '@hooks';
+
 import { TodoItem } from '../Item';
 
 import style from './TodoList.module.scss';
 
 export const TodoList: FC = () => {
   const { todoList } = useAppSelector((state) => state.todo);
 
+  const { onSetDone } = useTodoTask();
+
   if (!todoList.length) return null;
 
   return (
     <section className={style.container}>
       <ul className={style.list}>
-        {todoList.map(({ text, id, creationDate, expirationDate }) => (
+        {todoList.map(({ id, text, isDone, creationDate, expirationDate }) => (
           <li key={id}>
             <TodoItem
+              id={id}
               text={text}
+              isDone={isDone}
               creationDate={creationDate}
               expirationDate={expirationDate}
```

Comment:
Why don't you pass the item as one property?

Reply:
If I pass the item if 1 property, the memoization is not gonna work. Or you mean to spread the item to props?

Reply:
Changed TodoItem to be able to accept 'todo' prop as one property. Changed todoReducer,setTodoDone logic to fit the memoization.